include/openssl/hmac.h

```diff
@@ -250,12 +250,17 @@ OPENSSL_EXPORT int HMAC_CTX_copy(HMAC_CTX *dest, const HMAC_CTX *src);
 // Private functions
 typedef struct hmac_methods_st HmacMethods;
 
-// We use a union to ensure that enough space is allocated and never actually bother with the named members.
+// We use a union to ensure that enough space is allocated and never actually
+// bother with the named members. We do not externalize SHA3 ctx definition,
+// so hard-code ctx size below and use a compile-time assertion where that ctx
+// is defined to ensure it does not exceed size bounded by |md_ctx_union|. This
+// is OK because union members are never referenced, they're only used for sizing.
 union md_ctx_union {
   MD5_CTX md5;
   SHA_CTX sha1;
   SHA256_CTX sha256;
   SHA512_CTX sha512;
+  uint8_t sha3[400];
 };
```

Comment:
A bit of a bolder change, but could we move the `md_ctx_union` union to our internal headers? I don't see any existing OpenSSL reference to the union. That way we could abstract `400` to a `#define` variable instead of a hard-coded number.

If we're not comfortable doing so, we can leave this as is though.